### PR TITLE
Improve orchestration speed and reliability

### PR DESCRIPTION
**Pull request by turntable-justin**
- Only pull subtasks (slow query) when looking at an individual run
- Removes n+1 query problem with artifacts (I think)
- Simplifies and speeds up task aggregation by setting up FK relationship with django-celery-results

Note:
- See turntable-so/django-celery-results for info on our custom fork. It involves very minor changes on a package that rarely updates -- last update was 2 months ago. Unfortunately there was no way to do this with inheritance alone.

----

> [!IMPORTANT]
> Optimize orchestration by improving task aggregation, updating serializers, and using a custom Celery backend.
> 
>   - **Behavior**:
>     - Optimize task aggregation by setting up FK relationship with `django-celery-results` in `models.py`.
>     - Only pull subtasks when viewing an individual run in `orchestration_views.py`.
>     - Remove n+1 query problem with artifacts in `serializers.py`.
>   - **Serializers**:
>     - Add `DBTOrchestratorListSerializer` and `TaskResultWithSubtasksSerializer` in `serializers.py`.
>     - Modify `TaskResultSerializer` to include `job_id` and `result` parsing.
>   - **Models**:
>     - Change `ScheduledWorkflow.id` to `CharField` and remove `replacement_identifier` in `workflows.py`.
>     - Update `TaskArtifact.task` to `OneToOneField` in `workflows.py`.
>   - **Migrations**:
>     - Add migrations `0042_remove_scheduledworkflow_replacement_identifier.py`, `0043_alter_scheduledworkflow_id.py`, and `0044_alter_taskartifact_task.py`.
>   - **Settings**:
>     - Change `CELERY_RESULT_BACKEND` to `django_celery_results.backends:DatabaseBackend` in `settings.py`.
>   - **Dependencies**:
>     - Update `django-celery-results` to use a custom fork in `pyproject.toml` and `requirements.lock`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=turntable-so%2Fturntable&utm_source=github&utm_medium=referral)<sup> for e6535430190f30476514bf454e696edbe965aa13. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: backend/pyproject.toml [hunk 1]
```diff
@@ -83,7 +83,7 @@ dependencies = [
     "daphne>=4.1.2",
     "celery[redis,gevent]>=5.4.0",
     "flower>=2.0.1",
-    "django-celery-results>=2.5.1",
+    "django-celery-results @ git+https://github.com/turntable-so/django-celery-results",
```
Comment (ianrtracey) [inline backend/pyproject.toml:86]: we're using a fork of this now?

File: backend/api/serializers.py [hunk 2]
```diff
@@ -669,6 +685,76 @@ def _parse_meta(self, instance) -> dict | None:
         except (json.JSONDecodeError, AttributeError):
             return None
 
+    def _parse_json(self, data):
+        if not data:
+            return None
+        try:
+            return orjson.loads(data)
+        except orjson.JSONDecodeError:
```
Comment (ianrtracey) [inline backend/api/serializers.py:693]: I dont know if we ant to except here. A json parsing failure should just fail, no?

**Review by ianrtracey — APPROVED** [review]

File: backend/pyproject.toml (hunk 1 above)
Comment (turntable-justin) [reply backend/pyproject.toml:86]: Yes! To solve the issue we had, we needed to add a FK relation on the task result model, which we couldn't do with inheritance alone unfortunately. But the underlying repo is rarely updated, so we should be ok.

File: backend/api/serializers.py (hunk 2 above)
Comment (turntable-justin) [reply backend/api/serializers.py:693]: Agree with this. This is from an older era of the code when the typing of the result was less reliable. I'll make this change

**Pull request merged**